### PR TITLE
[#111] Json-RPC support

### PR DESCRIPTION
**Pull request by usmansaleem**
Expose JSON-RPC 2.0 service endpoint at /rpc/v1. 

Fix issue #111 

Sample calls:
https://github.com/PegaSysEng/eth2signer/issues/111#issuecomment-661445801

### CODE REVIEW

File: core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/exceptions/InvalidDataFormatException.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.eth2signer.core.service.jsonrpc.exceptions;
+
+import com.github.arteam.simplejsonrpc.core.annotation.JsonRpcError;
+
+@JsonRpcError(code = -30001, message = "Invalid data format. Not a hex string")
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/exceptions/InvalidDataFormatException.java:17]: Don't think we our own specific error, could this just be INVALID_PARAMS json RPC error?

File: core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/SigningService.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    try {
+      return signerForIdentifierList.stream()
+          .map(signerForIdentifier -> signerForIdentifier.sign(identifier, data))
+          .flatMap(Optional::stream)
+          .findFirst()
+          .orElseThrow(
+              () ->
+                  new SignerNotFoundException(
+                      "Signer not found for identifier", new String[] {data}));
+    } catch (final IllegalArgumentException e) {
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/SigningService.java:59]: Do unhandled exceptions return an INTERNAL_ERROR json RPC error?

File: acceptance-tests/src/test/java/tech/pegasys/eth2signer/tests/UpcheckAcceptanceTest.java [hunk 3]
```diff
@@ -35,4 +37,26 @@ void upcheckOnCorrectPortRespondsWithOK() {
...
+    final Response response =
+        given()
+            .baseUri(signer.getUrl())
+            .body("{\"jsonrpc\":\"2.0\",\"method\":\"upcheck\",\"params\":[],\"id\":1}")
+            .post(JSON_RPC_PATH);
+
+    validateBodyMatches(response);
+  }
+
+  private void validateBodyMatches(final Response response) {
```
[the remark this answers is not in the document]
Comment (usmansaleem) [reply acceptance-tests/src/test/java/tech/pegasys/eth2signer/tests/UpcheckAcceptanceTest.java]: renamed.

File: acceptance-tests/src/test/java/tech/pegasys/eth2signer/tests/publickeys/JsonRpcPublicKeysAcceptanceTest.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+import org.hamcrest.Matcher;
+import org.junit.jupiter.api.Test;
+
+public class JsonRpcPublicKeysAcceptanceTest extends PublicKeysAcceptanceTestBase {
+  @Test
```
[the remark this answers is not in the document]
Comment (usmansaleem) [reply acceptance-tests/src/test/java/tech/pegasys/eth2signer/tests/publickeys/JsonRpcPublicKeysAcceptanceTest.java]: I've deleted this file and combined both rpc and api test in one class (similar to signing).

File: core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/exceptions/InvalidDataFormatException.java (hunk 1 above)
Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/exceptions/InvalidDataFormatException.java]: Renamed the exception to use the "Invalid params" standard code.

File: core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/SigningService.java (hunk 2 above)
Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/SigningService.java:69]: yes, that is correct.

**Pull request merged**